hips/draw/simple.py

```diff
@@ -1,76 +1,89 @@
+class SimpleTilePainter:
+    """Paint HiPS tiles onto an all-sky image using a simple projective
+    transformation method. The algorithm implemented is described
+    here: :ref:`drawing_algo`.
 
     Parameters
     ----------
     geometry : `~hips.utils.WCSGeometry`
         An object of WCSGeometry
-    tiles : List[HipsTile]
-        A list of HipsTile
     hips_survey : `~hips.HipsSurveyProperties`
         HiPS survey properties
     tile_format : `str`
         Format of HiPS tile
 
-    Returns
-    -------
-    np.ndarray
-        Returns a numpy array containing all HiPS tiles projected onto it
-    """
-    if tile_format == 'jpg':
-        shape = (geometry.shape.height, geometry.shape.width, 3)
-    elif tile_format == 'png':
-        shape = (geometry.shape.height, geometry.shape.width, 4)
-    else:
-        shape = (geometry.shape.height, geometry.shape.width)
-    image = np.zeros(shape)
-    for tile in tiles:
-        painter = SimpleTilePainter(geometry, hips_survey, tile)
-        image += painter.warp_image()
-    return image
-
-
-class SimpleTilePainter:
-    """Paint a single tile using a simple projective transformation method.
-    The algorithm implemented is described here: :ref:`drawing_algo`.
-
-    Parameters
-    ----------
-    geometry : `~hips.utils.WCSGeometry`
-        An object of WCSGeometry
-    hips_survey : `~hips.HipsSurveyProperties`
-        HiPS survey properties
-    tile : `HipsTile`
-       An object of HipsTile
+    Examples
+    --------
+    >>> from hips.utils import WCSGeometry
+    >>> from hips.draw import SimpleTilePainter
+    >>> from hips.tiles import HipsSurveyProperties
+    >>> from astropy.coordinates import SkyCoord
+    >>> url = 'http://alasky.unistra.fr/DSS/DSS2Merged/properties'
+    >>> hips_survey = HipsSurveyProperties.fetch(url)
+    >>> geometry = WCSGeometry.create_simple(
+    ...     skydir=SkyCoord(0, 0, unit='deg', frame='icrs'),
+    ...     width=2000, height=1000, fov="3 deg",
+    ...     coordsys='icrs', projection='AIT'
+    ... )
+    >>> simple_tile_painter = SimpleTilePainter(geometry, hips_survey, 'fits')
+    >>> simple_tile_painter.hips_order
+    7
+    >>> simple_tile_painter.run()
+    >>> simple_tile_painter.image[0]
+    [[ 104.0982597    62.56927738   52.15683176]
+     [ 107.48257778   63.78870903   53.12130847]
+     [ 110.50075449   64.98920598   53.02016096]
+     ...,
+     [ 112.695798     51.03053188   62.20767366]
+     [ 118.12587546   57.11817351   56.96976768]
+     [ 124.43421962   68.33086816   49.93003909]]
     """
 
-    def __init__(self, geometry: WCSGeometry, hips_survey: HipsSurveyProperties, tile: HipsTile) -> None:
+    def __init__(self, geometry: WCSGeometry, hips_survey: HipsSurveyProperties, tile_format: str) -> None:
         self.geometry = geometry
         self.hips_survey = hips_survey
-        self.tile = tile
+        self.tile_format = tile_format
+
+    @property
+    def hips_order(self) -> int:
```

Comment:
Suggest to rename this to the more specific term `draw_hips_order` to avoid confusion with e.g. the `hips_survey.hips_order`.